Created two new patterns activemq-private-lambda-java-sam and rabbitmq-private-lambda-java-sam (#2850)
aws-samples/serverless-patterns: activemq-private-lambda-java-sam/README.md

# Java AWS Lambda ActiveMQ (in private subnets) consumer, using AWS SAM

> This pattern is an example of a Lambda function written in Java that consumes messages from Amazon MQ (Apache ActiveMQ). The pattern demonstrates how a SAM project can be configured to deploy an AWS Lambda function written in Java with an Amazon MQ (Apache ActiveMQ) event trigger. The pattern demonstrates how Amazon MQ (Apache ActiveMQ) messages can be parsed in an AWS Lambda function and the parsed content output to an Amazon DynamoDB table. The pattern provides an AWS CloudFormation template to install and set-up an Amazon MQ (Apache ActiveMQ) cluster inside private subnets in an Amazon VPC. The CloudFormation template also installs an Amazon EC2 instance with tools necessary to configure the Amazon MQ (Apache ActiveMQ) cluster to generate Amazon MQ (Apache ActiveMQ) messages and configuration needed by a Java producer that can generate the Amazon MQ (Apache ActiveMQ) messages.

Reviewer: ```suggestion
This pattern is an example of a Lambda function written in Java that consumes messages from Amazon MQ (Apache ActiveMQ), located in an private subnet. The function parses the ActiveMQ messages and stores the results in an Amazon DynamoDB table. The pattern provides an AWS CloudFormation template to install and set-up an Amazon MQ (Apache ActiveMQ) cluster inside private subnets in an Amazon VPC. The CloudFormation template also launches an Amazon EC2 instance with tools necessary to configure the Amazon MQ (Apache ActiveMQ) cluster and generate Amazon MQ (Apache ActiveMQ) messages.
```

* [Connect to the EC2 instance] - Once the AWS CloudFormation stack is created, you can go to the EC2 console and log into the machine using either "Connect using EC2 Instance Connect" or "Connect using EC2 Instance Connect Endpoint" option under the "EC2 Instance Connect" tab. In case you are using SSM Instance connect, you are not initially placed in the home directory. If you connect as ssm-user, you need to sudo su to ec2-user for this to work.
Note: You may need to wait for some time after the CloudFormation stack is created, as some UserData scripts continue running post creation.

## Pre-requisites to Deploy the sample Lambda function

The EC2 instance that was created by running the AWS CloudFormation template has all the software that will be needed to deploy the Lambda function.

The AWS SAM CLI is a serverless tool for building and testing Lambda applications.

> * Java - On the EC2 instance, we have installed the version of Java that you selected. We have installed Amazon Corrretto JDK of the version that you had selected at the time of specifying the input parameters in the Cloudformation template. At the time of publishing this pattern, only Java versions 11, 17 and 21 are supported by AWS SAM

Reviewer: ```suggestion
* Java - On the EC2 instance, we installed the version of Java and Amazon Corretto JDK you selected at deployment. At the time of publishing this pattern, only Java versions 11, 17 and 21 are supported by AWS SAM
```

chmod +x commands.sh
```

You should see a script called commands.sh. Run that script by passing a random string and a number between 1 and 500. Either send at least 10 messages or wait for 300 seconds (check the values of BatchSize: 10 and MaximumBatchingWindowInSeconds: 300 in the template.yaml file)

```
[ec2-user@ip-10-0-0-126 ~]$ sh ./commands.sh firstBatch 10
```

You should see output similar to what is shown below:

Sent out one message - Number 1 at time = 1760937987906
Sent out one message - Number 2 at time = 1760937987909
Sent out one message - Number 3 at time = 1760937987910
Sent out one message - Number 4 at time = 1760937987911
Sent out one message - Number 5 at time = 1760937987913
Sent out one message - Number 6 at time = 1760937987914
Sent out one message - Number 7 at time = 1760937987915
Sent out one message - Number 8 at time = 1760937987916
Sent out one message - Number 9 at time = 1760937987918
Sent out one message - Number 10 at time = 1760937987919

Once the messages have been sent, check Amazon CloudWatch logs and you should see messages for the Amazon CloudWatch Log Group with the name of the deployed Lambda function.

> When you run the above script, it sends messages with JSON records to the Amazon MQ (Apache ActiveMQ) cluster on the queue on which the lambda function is listening on. The lambda function listens on the published Amazon MQ (Apache ActiveMQ) messages on the queue.

Reviewer: ```suggestion
When you run the above script, it sends messages with JSON records to the Amazon MQ (Apache ActiveMQ) cluster on the queue on which the lambda function is listening on. The Lambda function listens on the published Amazon MQ (Apache ActiveMQ) messages on the queue.
```
